### PR TITLE
DETR XAI

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

Enhance XAI capabilities for RTDETR and D-Fine models by implementing support for saliency map generation and feature vector outputs, compatibility with both PyTorch and OpenVINO frameworks.

Person Saliency Map:
000000046872_class_0_overlay

Car Saliency Map:
000000046872_class_1_overlay

### How to test

`otx explain --work_dir otx-workspace --dump True --explain_config.postprocess True`

### Checklist

- [x] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [x] I have ran e2e tests and there is no issues.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [x] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by sovrasov** [comment]
@eugene123tw thanks for the PR! Could you attach some explain results here for a reference?

File: src/otx/algo/detection/d_fine.py [hunk 1]
```diff
@@ -306,3 +336,35 @@ def _optimization_config(self) -> dict[str, Any]:
...
+        entity: DetBatchDataEntity,
+        mode: str = "tensor",  # noqa: ARG004
+    ) -> dict[str, torch.Tensor]:
+        """Forward function for explainable detection model."""
+        backbone_feats = self.encoder(self.backbone(entity.images))
+        predictions = self.decoder(backbone_feats, explain_mode=True)
+
+        feature_vector = self.feature_vector_fn(backbone_feats)
+
+        splits = [f.shape[-2] * f.shape[-1] for f in backbone_feats]
```
Comment (sovrasov) [inline src/otx/algo/detection/d_fine.py]: It looks like snippet
```python
splits = [f.shape[-2] * f.shape[-1] for f in backbone_feats]
# Permute and split logits in one line
raw_logits = torch.split(predictions["raw_logits"].permute(0, 2, 1), splits, dim=-1)

# Reshape each split in a list comprehension
raw_logits = [
    logits.reshape(f.shape[0], -1, f.shape[-2], f.shape[-1])
    for logits, f in zip(raw_logits, backbone_feats)
]
```
is repeated 3 or 4 times in the PR. Is there any way to generalize it?

File: src/otx/algo/detection/detectors/detection_transformer.py [hunk 2]
```diff
@@ -96,18 +95,28 @@ def export(
...
+            splits = [f.shape[-2] * f.shape[-1] for f in backbone_feats]
+            # Permute and split logits in one line
+            raw_logits = torch.split(predictions["raw_logits"].permute(0, 2, 1), splits, dim=-1)
+
+            # Reshape each split in a list comprehension
+            raw_logits = [
+                logits.reshape(f.shape[0], -1, f.shape[-2], f.shape[-1])
+                for logits, f in zip(raw_logits, backbone_feats)
+            ]
+            saliency_map = self.explain_fn(raw_logits)
```
Comment (sovrasov) [inline src/otx/algo/detection/detectors/detection_transformer.py:108]: What is the `explain_fn` here? Vit-ReciproCAM?

File: src/otx/algo/detection/d_fine.py (hunk 1 above)
Comment (eugene123tw) [reply src/otx/algo/detection/d_fine.py]: I created a staticmethod (`DETR.split_and_reshape_logits`) and reuse this method instead.

File: src/otx/algo/detection/detectors/detection_transformer.py (hunk 2 above)
Comment (eugene123tw) [reply src/otx/algo/detection/detectors/detection_transformer.py:108]: No, it's the default DetClassProbabilityMap. I updated the visual result in the description.

**Review by sovrasov — APPROVED** [review]

**Review by negvet — APPROVED** [review]
As @sovrasov noted, `DetClassProbabilityMap` has some normalization issues.
I mean it is quite sensitive to the particular model and particular dataset.
Galina tried to come up with some uniformly good solution, but this was not that successful.
The result seems not perfect, but probably acceptable. Assuming that the model is well trained.

For the future, I would propose to revisit `DetClassProbabilityMap`. Basically, it is just visualization of the classification head output. Probably some modification or other methods might work better to explain object detection models.

**Review by Daankrol — APPROVED** [review]
